integrations/common/http.go

```diff
@@ -19,30 +20,84 @@ const (
 	ContentTypeForm            = "application/x-www-form-urlencoded"
 	ContentTypeJSON            = "application/json"                // Accept
 	ContentTypeJSONCharsetUTF8 = "application/json; charset=utf-8" // Content-Type
+
+	HTTPTimeout = 3 * time.Second
+	HTTPMaxSize = 1 << 23 // 2^23 bytes = 8 MiB
 )
 
+func HTTPDeleteJSON(ctx context.Context, u, auth string, payload any) ([]byte, error) {
+	if s, ok := payload.(string); ok {
```

Comment:
This is risky, I don't like guessing.
I prefer to explicit declare that you want to convert the value into JSON. Maybesome something in the lines of:
```go
type JSON struct {
	v any
}

func JSONed(v any) JSON {
	return JSON{v}
}

func encode(v any) {
	if _, ok := v.(JSON); ok {
		println("json")
	} else {
		println("not json")
	}
}

func main() {
	encode("a")
	encode(JSONed("a"))
}
```

Or pass `io.Reader`

Reply:
That's a whole lot of code to replace a simple "if string return bytes, else convert to JSON".